pkg/controller/beat/common/config.go

```diff
@@ -13,66 +13,92 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	beatv1beta1 "github.com/elastic/cloud-on-k8s/pkg/apis/beat/v1beta1"
-	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/association"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common/reconciler"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common/settings"
 	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
 )
 
-// setOutput will set the output section in Beat config according to the association configuration.
-func setOutput(cfg *settings.CanonicalConfig, client k8s.Client, associated commonv1.Association) error {
+type DefaultConfig struct {
+	// Managed config is handled by ECK and will not be nullified by user provided config.
+	Managed *settings.CanonicalConfig
+	// Unmanaged config is default config that will be overridden by any user provided config.
+	Unmanaged *settings.CanonicalConfig
+}
+
+// buildOutputConfig will create the output section in Beat config according to the association configuration.
+func buildOutputConfig(client k8s.Client, associated beatv1beta1.BeatESAssociation) (*settings.CanonicalConfig, error) {
 	if !associated.AssociationConf().IsConfigured() {
-		return nil
+		return settings.NewCanonicalConfig(), nil
 	}
 
-	username, password, err := association.ElasticsearchAuthSettings(client, associated)
+	username, password, err := association.ElasticsearchAuthSettings(client, &associated)
 	if err != nil {
-		return err
+		return settings.NewCanonicalConfig(), err
 	}
 
-	esOutput := settings.MustCanonicalConfig(
-		map[string]interface{}{
-			"output.elasticsearch": map[string]interface{}{
-				"hosts":    []string{associated.AssociationConf().GetURL()},
-				"username": username,
-				"password": password,
-			},
-		})
-
-	if err := cfg.MergeWith(esOutput); err != nil {
-		return err
+	esOutput := map[string]interface{}{
```

Comment:
Doesn't need to be resolved in your change, but this made me realize we need to add documentation on how beats and ssl verification interacts with custom certs. Opened this issue:
https://github.com/elastic/cloud-on-k8s/issues/3216

The only way I can think of being smarter here is if we check that the current associated certificate is valid for the host name the beat will use, and if not, setting ssl verification mode to `none`. I'm not sure we want to do that silently though, and probably want the users to know they are disabling ssl.

I do wonder if we want to fail the association if we know that the beat cannot actually associate successfully with the current certificate.